pandas/tools/merge.py

```diff
@@ -1202,16 +1202,11 @@ def _sort_labels(uniques, left, right):
         # tuplesafe
         uniques = Index(uniques).values
 
-    sorter = uniques.argsort()
+    l = len(left)
+    labels = np.concatenate([left, right])
```

Comment:
Not related with `difference` or `symmetric_difference` but may be with `union`.

I think that issue is due to [`indexes.api._union_indexes`](https://github.com/pydata/pandas/blob/master/pandas/indexes/api.py#L45), which shows an inconsistent behaviour: it looks like it sorts almost always (even a single index) except if all passed indexes are equal.